recipes/full_finetune_distributed.py

```diff
@@ -722,6 +775,27 @@ def train(self) -> None:
 
         self._profiler.stop()
 
+    def get_lr_scheduler(self) -> float:
```

Comment:
If we are gonna have this as a utility I would take it out of this recipe since it's equally applicable to the single-device case. Also I wouldn't call it `get_lr_scheduler`, since we're really getting the current lr, not the scheduler itself.

Reply:
Single_device get lr with the assumption that [all optimizers have the same LR.](https://github.com/pytorch/torchtune/blob/5de500147e57d4d35e296faba68b296c03a1a2a0/recipes/full_finetune_single_device.py#L659)

Shall we apply the logic here to single device, rename to `get_lr` and move to `/torchtune/utils`?
or just taking the same assumption as single device?

Reply:
Technically I think this was also previously the assumption in this recipe, right? Since we just log the LR from the first param group. So we should be able to maintain the same logic for both